src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,48 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.query.Query;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDB = userService.findByEmail(email);
+        if (userFromDB.isEmpty() || !userFromDB.get().getPassword().equals(HashUtil
+                .hashPassword(password, userFromDB.get().getSalt()))) {
+            throw new AuthenticationException("Incorrect login or password");
+        }
+        return userFromDB.get();
+    }
+
+    @Override
+    public User register(String email, String password) throws RegistrationException {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```

Comment:
This is Service class. We should use DAO or **another Service** (in this case) if need DB data and we can not open Session in Service.